go-controller/pkg/ovn/gress_policy.go

```diff
@@ -382,79 +388,38 @@ func (gp *gressPolicy) modifyACLAllow(oldMatch, newMatch string) error {
 	return nil
 }
 
-func constructIPStringForACL(direction string, ipCIDRs []string) string {
+func constructIPStringForACL(direction string, ipBlock *ipBlock) string {
 	var v4MatchStr, v6MatchStr, matchStr string
 	v4CIDRs := make([]string, 0)
 	v6CIDRs := make([]string, 0)
 
-	for _, cidr := range ipCIDRs {
+	if utilnet.IsIPv6CIDRString(ipBlock.cidr) {
+		matchStr = fmt.Sprintf("%s.%s == %s", "ip6", direction, ipBlock.cidr)
+	} else {
+		matchStr = fmt.Sprintf("%s.%s == %s", "ip4", direction, ipBlock.cidr)
+	}
+
+	for _, cidr := range ipBlock.except {
 		if utilnet.IsIPv6CIDRString(cidr) {
 			v6CIDRs = append(v6CIDRs, cidr)
 		} else {
 			v4CIDRs = append(v4CIDRs, cidr)
 		}
 	}
+
 	if len(v4CIDRs) > 0 {
 		v4AddressSetStr := strings.Join(v4CIDRs, ", ")
-		v4MatchStr = fmt.Sprintf("%s.%s == {%s}", "ip4", direction, v4AddressSetStr)
-		matchStr = v4MatchStr
+		v4MatchStr = fmt.Sprintf("%s.%s != {%s}", "ip4", direction, v4AddressSetStr)
+		matchStr = fmt.Sprintf("%s && %s", matchStr, v4MatchStr)
```

Comment:
should we need parenthesis to concatenate expressions?

Reply:
it can make the logic easier to read in my opinion